sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -18,208 +18,170 @@
+        .then((response: HttpResponse<gapi.client.bigquery.ListTablesResponse>) => {
+          console.log('== tables: ', response);
+          const tableResults: Result[] = response.result.tables.map(this._bqTableToResult.bind(this)) as Result[];
+          resultHandler(tableResults);
+        });
+  }
+
+  _bqProjectToResult(bqProject: gapi.client.bigquery.ProjectResource): Result {
+    return {
+      name: bqProject.id,
+      type: 'project',
+    } as Result;
   }
 
-  /** Reads the user's query values, queries for tables, and updates the results. */
-  _search() {
-    this._collectionsList = this._generateFakeCollectionsListForTesting();
-    this._debugCallBigQuery();   // For debugging, make some BigQuery calls
+  _bqDatasetToResult(bqDataset: gapi.client.bigquery.DatasetResource): Result {
+    return {
+      name: bqDataset.id,
+      type: 'dataset',
+    } as Result;
   }
 
-  // Make some calls to the BigQuery API and log the results to the console.
-  _debugCallBigQuery() {
-    const sampleProject = 'bigquery-public-data';
-    const sampleDataset = 'samples';
-    const emptyFilter = '';
-    GapiManager.listBigQueryProjects()
-        .then((response: HttpResponse<gapi.client.bigquery.ListProjectsResponse>) => console.log('== projects: ', response));
-    GapiManager.listBigQueryDatasets(sampleProject, emptyFilter)
-        .then((response: HttpResponse<gapi.client.bigquery.ListDatasetsResponse>) => console.log('== datasets: ', response));
-    GapiManager.listBigQueryTables(sampleProject, sampleDataset, emptyFilter)
-        .then((response: HttpResponse<gapi.client.bigquery.ListTablesResponse>) => console.log('== tables: ', response));
+  _bqTableToResult(bqTable: gapi.client.bigquery.TableResource): Result {
+    return {
+      name: bqTable.id,
+      type: 'table',
+    } as Result;
   }
 
   /**
-   * Creates a new ItemListRow object for each entry in the collections list, and sends
+   * Creates a new ItemListRow object for each entry in the results list, and sends
    * the created list to the item-list to render.
    */
-  _renderCollectionsList() {
-    this.$.collections.rows = this._collectionsList.map((collection) => {
+  _renderResultsList() {
+    this.$.results.rows = this._resultsList.map((result) => {
       return {
-        firstCol: collection.name,
-        icon: 'folder',
-        secondCol: collection.status,
+        firstCol: result.name,
+        icon: this._typeToIcon(result.type),
+        secondCol: result.type,
         selected: false,
       };
     });
   }
 
-  _showTablesForCollection(collection: Collection) {
-    console.log('== collection selected:', collection);
-    if (!collection.tables) {
-      collection.tables = this._generateFakeTablesListForTesting();
-    }
-    this._tablesList = collection.tables;
-  }
-
-  _clearTablesList() {
-    this._tablesList = [];
+  _typeToIcon(type: string): string {
```

Comment:
If we put each type's icon in its `_<type>ToResult` function we don't have to worry about changes to those types later on.